hoon: virtualization core; namespacing

Quodss opened the pull request:
Adds `+vi` door to `hoon.hoon` with default sample being a pass-through scry gate. To call e.g. `+mole` with empty namespace:

```
(~(mole vi ~) t)
```

Breaking changes: namespacing of virtualization gates. PR includes changes to the codebase for compatibility

mikolajpp commented on pkg/base-dev/lib/bip39/english.hoon:
```diff
@@ -1163,7 +1163,7 @@
   "much"
   "muffin"
-  "mule"
+  "mule:vi"
```
This sneaked in here by mistake.

Quodss commented:
CI might OOM because mole/mule are not jetted here, leading to kicking traps twice instead of once. https://github.com/urbit/vere/pull/814 addresses this

Quodss commented:
It just occured to me that making this change would induce jet mismatch in mole/mule. Indeed, jets assume that the results of kicking a trap in the current virtualization scope and running it virtualized with pass-through scry gates would return the same result if the product of the virtual kick is succesful. That would also be true for an empty scry gate, but not for all scry gates in general, as the results of scries might differ. I'll change door sample to be a loobean instead

Quodss commented:
removed namespacing changes and added aliases instead.

Weird behavior: this memoizes:
```
> =t1 |.(~>(%memo./a ~&(%foo ~)))
%foo
> (~(mole vi |) t1)
[~ ~]
> (~(mole vi |) t1)
[~ ~]
```

this doesn't:

```
> (~(mole vi |) |.(~>(%memo./a ~&(%foo ~))))
[~ ~]
%foo
> (~(mole vi |) |.(~>(%memo./a ~&(%foo ~))))
[~ ~]
```

joemfb commented on pkg/arvo/sys/hoon.hoon:
```diff
@@ -6310,6 +6269,72 @@
+      %mure  mure
+      %mute  mute
+    ==
+  |%
```
Style nit: this can just be `|_  for=?`, and the parent core can be specified by wing: `~%  %vi  ..vi`.

Quodss commented:
this is embrassing... but now it at least fails for the right reason (OOM)
